lib/rules/valid-template-root.js

```diff
@@ -39,13 +39,49 @@ module.exports = {
           }
         }
 
-        if (hasSrc && rootElements.length > 0) {
+        if (rootElements.length > 0) {
+          if (hasSrc) {
+            for (const element of rootElements) {
+              context.report({
+                node: element,
+                loc: element.loc,
+                message:
+                  "The template root with 'src' attribute is required to be empty."
+              })
+            }
+          }
+
+          let hasRootVIfLength = 0
+          let hasRootVElseLength = 0
+          let hasRootVElseIfLength = 0
+
           for (const element of rootElements) {
+            if (element.type === 'VElement') {
+              const hasRootVIf = utils.hasDirective(element, 'if')
+              const hasRootVElse = utils.hasDirective(element, 'else')
+              const hasRootVElseIf = utils.hasDirective(element, 'else-if')
+              if (hasRootVIf) {
+                hasRootVIfLength += 1
+              }
+              if (hasRootVElse) {
+                hasRootVElseLength += 1
+              }
+              if (hasRootVElseIf) {
+                hasRootVElseIfLength += 1
+              }
+            }
+          }
```

Comment:
Could be simplified:

```suggestion
          const hasRootVIfLength = rootElements.filter(element => element.type === 'VElement' && utils.hasDirective(element, 'if')).length
          const hasRootVElseLength = rootElements.filter(element => element.type === 'VElement' && utils.hasDirective(element, 'else')).length
          const hasRootVElseIfLength = rootElements.filter(element => element.type === 'VElement' && utils.hasDirective(element, 'else-if')).length
```